File: 2labz/2labz2/2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func normalizeEmail(email string) string {
	email = strings.TrimSpace(email)
	parts := strings.Split(email, "@")
	if len(parts) != 2 {
		return "" // пропускаем некорректные email
	}
	username := strings.ReplaceAll(parts[0], ".", "")
	username = strings.Split(username, "+")[0]
	return username + "@" + parts[1]
}

func main() {
	fmt.Println("Введите email-адреса через запятую и пробел:")
	reader := bufio.NewReader(os.Stdin)
	input, _ := reader.ReadString('\n')
	input = strings.TrimSpace(input)

	emails := strings.Split(input, ", ")
	uniqueEmails := make(map[string]bool)

	for _, email := range emails {
		normalized := normalizeEmail(email)
		if normalized != "" {
			uniqueEmails[normalized] = true
		}
	}

	fmt.Println("\nУникальные адреса после нормализации:")
	for email := range uniqueEmails {
		fmt.Println("-", email)
	}

	fmt.Printf("\nКоличество уникальных адресов: %d\n", len(uniqueEmails))
}

File: 2labz/2labz2/2.swift
import Foundation

func normalizeEmail(_ email: String) -> String {
    let parts = email.components(separatedBy: "@")
    var username = parts[0].replacingOccurrences(of: ".", with: "")
    if let plusRange = username.range(of: "+") {
        username = String(username[..<plusRange.lowerBound])
    }
    return "\(username)@\(parts[1])"
}

print("Введите email-адреса через запятую и пробел:")
if let input = readLine() {
    let emails = input.components(separatedBy: ", ")
    var uniqueEmails = Set<String>()

    for email in emails {
        uniqueEmails.insert(normalizeEmail(email.trimmingCharacters(in: .whitespaces)))
    }

    print("\nУникальные адреса после нормализации:")
    for email in uniqueEmails {
        print("- \(email)")
    }

    print("\nКоличество уникальных адресов: \(uniqueEmails.count)")
}
